Pages: Add notice for homepage editing

abotic opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

Fixes:
 - https://github.com/Automattic/wp-calypso/issues/98800

## Proposed changes:

This PR adds a notice to the Pages list screen that helps users find and edit their homepage when it's controlled by a block theme template rather than being a specific page. This addresses user confusion about how to edit their homepage when it's set to display latest posts.

Key features:
* Simple card with clear message "Your homepage is set to display latest posts"
* Direct "Edit homepage" link to the Site Editor
* Only appears for block themes with homepage set to show latest posts
* Positioned consistently in the Pages list
* Responsive design that works on mobile

## Implementation details:
* Uses clean, minimal JavaScript to insert the card in the right position
* Follows WordPress coding standards and accessibility best practices
* Only loads on the Pages screen when relevant homepage conditions are met
* Preserves the existing page listing functionality

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:

1. Apply the PR to a site using a block theme (like Twenty Twenty-Four)
2. Make sure your site homepage is set to "Your latest posts" in Settings > Reading
3. Go to Pages in wp-admin
4. Verify you see the notice card appearing above the pages list filters
5. Check that the "Edit homepage" link takes you to the Site Editor
6. Verify the notice doesn't appear if:
   - You're using a classic theme
   - You have a static page set as homepage
   - You're on a different admin screen

Before | After
-------|------
![image](https://github.com/user-attachments/assets/53eafa4b-63a2-407f-a15e-90dc20c00a59) | ![image](https://github.com/user-attachments/assets/c05a7b0a-ca3e-4c50-b4df-009419ca210e)

abotic committed:
Pages: Add notice for homepage editing

abotic committed:
changelog

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+
+	wp_register_style( 'wpcom-homepage-notice', false, array(), '20250310' );
+	wp_enqueue_style( 'wpcom-homepage-notice' );
+	wp_add_inline_style(
```
What do you think of moving the JS and CSS to their own files instead of having them inlined here?

Then you could use:
`jetpack_mu_wpcom_enqueue_assets( 'pages-homepage-notice', array( 'js', 'css' ) );`
To load them and versions and everything else gets handled for you.

abotic replied:
Had some issue using `jetpack_mu_wpcom_enqueue_assets` with this specific case, but I have moved JS and CSS to their own files.

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Homepage notice for the Pages list.
```
Maybe we can change the filename `homepage-notice.php` to something more meaningful. With the current naming, it isn't clear what the notice is about.

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+	}
+
+	$show_on_front  = get_option( 'show_on_front' );
+	$front_page_id  = get_option( 'page_on_front' );
```
Is the value returned here really an id?

abotic replied:
Yes, and it is used on other places in codebase with same variable name.
Sometimes, it is casted (int) for clarity which I did here in new commit.

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+			var noticeHTML = '<div class="wpcom-homepage-notice">' +
+				'<div class="wpcom-homepage-notice-content">' +
+					'<div class="wpcom-homepage-notice-info">' +
+						'<p class="wpcom-homepage-notice-text"><?php echo esc_js( $display_text ); ?></p>' +
```
Wouldn't `esc_html__` be better for this case than `esc_js`?

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+					'</div>' +
+					<?php if ( $can_edit ) : ?>
+					'<div class="wpcom-homepage-notice-actions">' +
+						'<a href="<?php echo esc_js( $edit_link ); ?>" class="wpcom-homepage-notice-edit-link">' +
```
Wouldn't `esc_attr__` be better for this case than `esc_js`?

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+					<?php if ( $can_edit ) : ?>
+					'<div class="wpcom-homepage-notice-actions">' +
+						'<a href="<?php echo esc_js( $edit_link ); ?>" class="wpcom-homepage-notice-edit-link">' +
+							'<?php echo esc_js( __( 'Edit homepage', 'jetpack-mu-wpcom' ) ); ?>' +
```
Wouldn't `esc_html__` be better for this case than `esc_js`?

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+			?>
+		<script>
+		(function() {
+			var noticeHTML = '<div class="wpcom-homepage-notice">' +
```
What do you think of creating the HTMLElement programmatically?

xavier-lc commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+            }
+        }
+    '
+	);
```
Wdyt about moving the css to a proper css file and then loading it with:

`wp_add_inline_style( 'jetpack-global-styles-frontend-style', file_get_contents( plugin_dir_path( __FILE__ ) . 'static/style.css', true ) );`

It would make the file more readable. Something like what's done on https://github.com/Automattic/jetpack/blob/trunk/projects/packages/jetpack-mu-wpcom/src/features/jetpack-global-styles/class-global-styles.php#L445

abotic replied:
Thanks, I didn't exactly follow the syntax as Richard also suggested something similar.
But I made sure to properly move the code in their own js/css/php files to make it cleaner.

xavier-lc commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+    '
+	);
+
+	add_action(
```
Similar here, it'd be great if we could move all this js to a separate file.

xavier-lc commented on projects/packages/jetpack-mu-wpcom/src/features/pages/homepage-notice.php:
```diff
@@ -0,0 +1,120 @@
+	);
+}
+
+add_action( 'admin_init', 'wpcom_add_homepage_notice' );
```
I couldn't see the notice until I made this change:

```suggestion
add_action( 'current_screen', 'wpcom_add_homepage_notice' );
```

[This comment](https://developer.wordpress.org/reference/functions/get_current_screen/#comment-3186) took me on the right path.

abotic replied:
That's a nice find! While it worked for me locally, I can see why you might had some issues. 
Updated the code.

abotic committed:
Fixing the comments
- Create separate CSS and JS files for better organization
- Use proper asset loading with direct file paths
- Improve element creation with programmatic DOM building
- Use current_screen hook for better timing
- Resolve other minor issues

abotic committed:
revert deleted dir

abotic committed:
final fixes

abotic committed:
Merge remote-tracking branch 'origin' into add/pages-homepage-notice

abotic commented:
@rcrdortiz @xavier-lc Thanks for the review and great feedback. 
I've addressed all comments and updated the code, so feel free to take another look at it.

rcrdortiz commented on projects/packages/jetpack-mu-wpcom/src/features/pages/pages-homepage-connection-banner.php:
```diff
@@ -0,0 +1,65 @@
+		'wpcom-pages-homepage-connection-banner',
+		plugin_dir_url( __FILE__ ) . 'css/pages-homepage-connection-banner.css',
+		array(),
+		'20250312'
```
No need to change, just so that you're aware it exists. For determining the version we could use the build version instead of hardcoding. Something like this:
````
$asset_file   = include Jetpack_Mu_Wpcom::BASE_DIR . "build/$asset_name/$asset_name.asset.php";
$version = $asset_file['version'] ?? '1.0.0';
````

abotic replied:
Oh, thanks for the tip. I was looking at other places in codebase and most of the code consisted of hardcoded date so I followed the same pattern. Will take that in mind for next time.

rcrdortiz reviewed (approved):
Works as expected for me. Thanks for all the changes!

xavier-lc reviewed (approved):
LGTM
